extras/kms/data_key_version_test.go

```diff
@@ -0,0 +1,296 @@
+				RootKeyVersionId: "root-key-version-id",
+			},
+			opType:          dbw.CreateOp,
+			wantErr:         true,
+			wantErrIs:       ErrInvalidParameter,
+			wantErrContains: "missing key",
+		},
+		{
+			name: "create-missing-data-key-id",
+			key: &dataKeyVersion{
+				PrivateId:        "private-id",
+				CtKey:            []byte("key"),
+				RootKeyVersionId: "root-key-version-id",
+			},
+			opType:          dbw.CreateOp,
+			wantErr:         true,
+			wantErrIs:       ErrInvalidParameter,
+			wantErrContains: "missing data key id",
+		},
+		{
+			name: "create-missing-root-key-version-id",
+			key: &dataKeyVersion{
+				PrivateId: "private-id",
+				CtKey:     []byte("key"),
+				DataKeyId: "data-key-id",
+			},
+			opType:          dbw.CreateOp,
+			wantErr:         true,
+			wantErrIs:       ErrInvalidParameter,
+			wantErrContains: "missing root key version id",
+		},
+		{
+			name: "update-immutable",
+			key: &dataKeyVersion{
+				PrivateId: "private-id",
+			},
+			opType:          dbw.UpdateOp,
+			wantErr:         true,
+			wantErrIs:       ErrInvalidParameter,
+			wantErrContains: "key is immutable",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			err := tc.key.vetForWrite(testCtx, tc.opType)
+			if tc.wantErr {
+				require.Error(err)
+				if tc.wantErrIs != nil {
+					assert.ErrorIs(err, tc.wantErrIs)
+				}
+				if tc.wantErrContains != "" {
+					assert.Contains(err.Error(), tc.wantErrContains)
+				}
+				return
+			}
+			require.NoError(err)
+		})
+	}
+}
+
+func TestDataKeyVersion_Encrypt(t *testing.T) {
+	t.Parallel()
+	const (
+		testKey = "test-key"
+	)
+	testCtx := context.Background()
+	testWrapper := wrapping.NewTestWrapper([]byte(defaultWrapperSecret))
+	tests := []struct {
+		name            string
+		key             *dataKeyVersion
+		wrapper         wrapping.Wrapper
+		wantErr         bool
+		wantErrIs       error
+		wantErrContains string
+	}{
+		{
+			name: "bad-cipher",
+			key: &dataKeyVersion{
+				Key: []byte(testKey),
+			},
+			wrapper:         aead.NewWrapper(),
+			wantErr:         true,
+			wantErrContains: "error wrapping value",
+		},
+		{
+			name: "missing-cipher",
+			key: &dataKeyVersion{
+				Key: []byte(testKey),
+			},
+			wantErr:         true,
+			wantErrContains: "missing cipher",
+		},
+		{
+			name: "success",
+			key: &dataKeyVersion{
+				Key: []byte(testKey),
+			},
+			wrapper: testWrapper,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			require.Empty(tc.key.CtKey)
+			require.NotEmpty(tc.key.Key)
+			err := tc.key.Encrypt(testCtx, tc.wrapper)
+			if tc.wantErr {
+				require.Error(err)
+				if tc.wantErrIs != nil {
+					assert.ErrorIs(err, tc.wantErrIs)
+				}
+				if tc.wantErrContains != "" {
+					assert.Contains(err.Error(), tc.wantErrContains)
+				}
+				return
+			}
+			require.NoError(err)
```

Comment:
As an extra sanity check, in the no-error case, can you also add an assert that the CtKey and Key are not equivalent? It's just a check in a non-structwrapping package that it is not goofing.

Reply:
ty